apps/server/src/modules/account/services/account-idm.service.ts

```diff
@@ -1,118 +1,117 @@
 import { ObjectId } from '@mikro-orm/mongodb';
 import { Injectable, NotImplementedException } from '@nestjs/common';
+import { ConfigService } from '@nestjs/config/dist/config.service';
 import { EntityNotFoundError } from '@shared/common';
 import { IdmAccount, IdmAccountUpdate } from '@shared/domain/interface';
 import { Counted, EntityId } from '@shared/domain/types';
 import { LegacyLogger } from '@src/core/logger';
-import { AccountIdmToDtoMapper } from '../mapper';
-import { AccountLookupService } from './account-lookup.service';
+import { Account, AccountSave } from '../domain';
+import { AccountIdmToDoMapper } from '../repo/mapper';
 import { AbstractAccountService } from './account.service.abstract';
-import { AccountDto, AccountSaveDto } from './dto';
+import { AccountConfig } from '../account-config';
 
 @Injectable()
 export class AccountServiceIdm extends AbstractAccountService {
 	constructor(
 		private readonly identityManager: IdentityManagementService,
-		private readonly accountIdmToDtoMapper: AccountIdmToDtoMapper,
-		private readonly accountLookupService: AccountLookupService,
+		private readonly accountIdmToDoMapper: AccountIdmToDoMapper,
 		private readonly idmOauthService: IdentityManagementOauthService,
-		private readonly logger: LegacyLogger
+		private readonly logger: LegacyLogger,
+		private readonly configService: ConfigService<AccountConfig, true>
 	) {
 		super();
 	}
 
-	async findById(id: EntityId): Promise<AccountDto> {
+	async findById(id: EntityId): Promise<Account> {
 		const result = await this.identityManager.findAccountById(id);
-		const account = this.accountIdmToDtoMapper.mapToDto(result);
+		const account = this.accountIdmToDoMapper.mapToDo(result);
 		return account;
 	}
 
-	async findMultipleByUserId(userIds: EntityId[]): Promise<AccountDto[]> {
+	// TODO: this needs a better solution. probably needs followup meeting to come up with something
+	async findMultipleByUserId(userIds: EntityId[]): Promise<Account[]> {
 		const results = new Array<IdmAccount>();
 		for (const userId of userIds) {
 			try {
 				// eslint-disable-next-line no-await-in-loop
 				results.push(await this.identityManager.findAccountByDbcUserId(userId));
 			} catch {
+				this.logger.error(`Error while searching for account with userId ${userId}`);
 				// ignore entry
 			}
 		}
-		const accounts = results.map((result) => this.accountIdmToDtoMapper.mapToDto(result));
+		const accounts = results.map((result) => this.accountIdmToDoMapper.mapToDo(result));
 		return accounts;
 	}
 
-	async findByUserId(userId: EntityId): Promise<AccountDto | null> {
+	async findByUserId(userId: EntityId): Promise<Account | null> {
 		try {
 			const result = await this.identityManager.findAccountByDbcUserId(userId);
-			return this.accountIdmToDtoMapper.mapToDto(result);
+			return this.accountIdmToDoMapper.mapToDo(result);
 		} catch {
+			this.logger.error(`Error while searching for account with userId ${userId}`);
```

Comment:
this.identityManager.findAccountByDbcUserId thow an error if no user is find? Than it look like all the other code should handle errors same for the account-db service. But logger.error is not avaible in logger, only in legacy logger and this should not be used anymore. The code not look well with catching error, logging and return null